inc/zoo/swar/SWAR.h

```diff
@@ -108,6 +110,17 @@ struct SWAR {
         return result;
     }
 
+    constexpr static auto evenLaneMask() {
+        using S = SWAR<NBits, T>;
+        static_assert(0 == S::Lanes % 2, "Only even number of elements supported");
+        using D = SWAR<NBits * 2, T>;
+        return S{(D::LeastSignificantBit << S::NBits) - D::LeastSignificantBit};
+    }
+
+    constexpr static auto oddLaneMask() {
+        return SWAR<NBits, T>{static_cast<T>(~evenLaneMask().value())};
```

Comment:
We must not add things that interfere with lane sizes that are not power-of-two bits.
I think we should discuss what is the convention we will use for the bits that are not a full lane.
I think it should be a user's choice, for example, "don't care", meaning the user tolerates any values in those bits, or "always-zero", that the library guarantees no operation will set those bits.